coralogix/resource_coralogix_globale_router.go

```diff
@@ -0,0 +1,723 @@
+	ID           types.String `tfsdk:"id"`
+	Name         types.String `tfsdk:"name"`
+	Description  types.String `tfsdk:"description"`
+	EntityType   types.String `tfsdk:"entity_type"`
+	Rules        types.List   `tfsdk:"rules"`    // RoutingRuleModel
+	FallBack     types.List   `tfsdk:"fallback"` // RoutingTargetModel
+	EntityLabels types.Map    `tfsdk:"entity_labels"`
+}
+
+type RoutingRuleModel struct {
+	Condition     types.String `tfsdk:"condition"`
+	Targets       types.List   `tfsdk:"targets"` // RoutingTargetModel
+	CustomDetails types.Map    `tfsdk:"custom_details"`
+	Name          types.String `tfsdk:"name"`
+}
+
+type RoutingTargetModel struct {
+	ConnectorId   types.String `tfsdk:"connector_id"`
+	PresetId      types.String `tfsdk:"preset_id"`
+	CustomDetails types.Map    `tfsdk:"custom_details"`
+}
+
+type ConfigOverridesModel struct {
+	MessageConfigFields   types.List `tfsdk:"message_config_fields"`   // MessageConfigFieldModel
+	ConnectorConfigFields types.List `tfsdk:"connector_config_fields"` // ConnectorOverrideFieldModel
+}
+
+type MessageConfigFieldModel struct {
+	FieldName types.String `tfsdk:"field_name"`
+	Template  types.String `tfsdk:"template"`
+}
+
+func (r *GlobalRouterResource) Metadata(_ context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_global_router"
+}
+
+func (r *GlobalRouterResource) Configure(_ context.Context, req resource.ConfigureRequest, resp *resource.ConfigureResponse) {
+	if req.ProviderData == nil {
+		return
+	}
+
+	clientSet, ok := req.ProviderData.(*clientset.ClientSet)
+	if !ok {
+		resp.Diagnostics.AddError(
+			"Unexpected Resource Configure Type",
+			fmt.Sprintf("Expected *clientset.ClientSet, got: %T. Please report this issue to the provider developers.", req.ProviderData),
+		)
+		return
+	}
+
+	r.client = clientSet.GetNotifications()
+}
+
+func (r *GlobalRouterResource) Schema(_ context.Context, _ resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		Version: 0,
+		Attributes: map[string]schema.Attribute{
+			"id": schema.StringAttribute{
+				Computed:    true,
+				Description: "The ID of the GlobalRouter.",
+			},
+			"name": schema.StringAttribute{
+				Required:    true,
+				Description: "Name of the GlobalRouter.",
+			},
+			"description": schema.StringAttribute{
+				Optional:    true,
+				Description: "Description of the GlobalRouter.",
+			},
+			"entity_type": schema.StringAttribute{
+				Required: true,
+				Validators: []validator.String{
+					stringvalidator.OneOf(validNotificationsEntityTypes...),
+				},
+				Description: "Type of the entity. Valid values are: " + strings.Join(validNotificationsEntityTypes, ", "),
+			},
+			"rules": schema.ListNestedAttribute{
+				Optional:    true,
+				Description: "Routing rules for the GlobalRouter.",
+				NestedObject: schema.NestedAttributeObject{
+					Attributes: map[string]schema.Attribute{
```

Comment:
Missing: custom_details (a string:string map) on the same level as name/condition